aikau/src/test/resources/alfresco/lists/AlfHashListTest.js

```diff
@@ -195,6 +196,37 @@ define(["alfresco/TestCommon",
 
          "Post Coverage Results": function() {
             TestCommon.alfPostCoverageResults(this, browser);
+         },
+
+         "Payload Data Loading is prevented when the hash vars are filter out": function() {
+            var badUrl = TestCommon.testWebScriptURL("/AlfHashListPayloadData#lib=Libraries&path=a/b/c"),
+                goodUrl = TestCommon.testWebScriptURL("/AlfHashListPayloadData#lib=Personal&path=d/e/f");
+
+            return browser.findByCssSelector("body")
+               .clearLog()
+               .end()
+
+            .get(badUrl)
+               .findByCssSelector("body")
+               .end()
+
+            .getLastPublish("ALF_RETRIEVE_DOCUMENTS_REQUEST")
+               .then(function(payload) {
+                  assert.notProperty(payload, "path", "There should be no path value in the payload");
+               })
+
+            .findByCssSelector("body")
+               .clearLog()
+               .end()
+
+            .get(goodUrl)
+               .findByCssSelector("body")
+               .end()
+
+            .getLastPublish("ALF_RETRIEVE_DOCUMENTS_REQUEST")
+               .then(function(payload) {
+                  assert.propertyVal(payload, "path", "d/e/f", "Path value incorrect");
+               });
          }
       };
```

Comment:
...and then add "Post Coverage Results (2") after the test here.

The reason being that we will get the coverage data for the code updates you've put (if we don't post coverage data before leaving a page then it is lost).